### PR TITLE
Read vcf utility

### PR DESCRIPTION
**Pull request by tcasstevens**

### CODE REVIEW

File: src/main/kotlin/biokotlin/util/VCFUtils.kt [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    headerLines.add(VCFInfoHeaderLine("ASM_Strand", 1, VCFHeaderLineType.String, "Assembly strand"))
+
+    return VCFHeader(headerLines, taxaNames)
+
+}
+
+/**
+ * Function to parse a VCF file into a map of ALT headers and a channel of SimpleVariant objects.
+ */
+fun parseVCFFile(filename: String): Pair<Map<String, AltHeaderMetaData>, Channel<Deferred<SimpleVariant>>> {
```
[the remark this answers is not in the document]
Comment (tcasstevens) [reply src/main/kotlin/biokotlin/util/VCFUtils.kt:154]: Yes, I need to add that. Wanted to get feedback on the design

[the remark this answers is not in the document]
Comment (zrm22) [reply src/main/kotlin/biokotlin/util/VCFUtils.kt:154]: I would recommend adding tests even if we decide to change the design.  It makes things go a lot more smoother if you test as you code up stuff, and you won't forget later to add in the test.

**Review by zrm22 — APPROVED** [review]

**Review by btmonier — APPROVED** [review]

File: src/main/kotlin/biokotlin/util/FileIO.kt [hunk 2]
```diff
@@ -23,4 +23,18 @@ fun bufferedReader(filename: String): BufferedReader {
...
+
+fun bufferedWriter(filename: String, append: Boolean = false): BufferedWriter {
+    return try {
+        if (filename.endsWith(".gz")) {
+            BufferedWriter(OutputStreamWriter(GZIPOutputStream(FileOutputStream(File(filename), append))))
+        } else {
+            BufferedWriter(OutputStreamWriter(FileOutputStream(File(filename), append)))
+        }
+    } catch (e: java.lang.Exception) {
+        println("bufferedWriter: problem getting writer for: $filename")
```
[the remark this answers is not in the document]
Comment (tcasstevens) [reply src/main/kotlin/biokotlin/util/FileIO.kt:36]: Do we have a standard logger for BioKotlin? It doesn't seem like it

File: src/main/kotlin/biokotlin/util/VCFUtils.kt [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                this.contig.compareTo(other.contig)
+            }
+        }
+    }
+
+    fun sample(index: Int) = samples[index]
+
+    fun genotype(sample: String): List<String> {
+        val sampleIndex = samples.indexOf(sample)
+        if (genotypes[sampleIndex].contains("|"))
```
[the remark this answers is not in the document]
Comment (tcasstevens) [reply src/main/kotlin/biokotlin/util/VCFUtils.kt]: Yes, thank you. Changed

File: src/main/kotlin/biokotlin/util/VCFUtils.kt [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+    }
+    return headerMetaData to channel
+
+}
+
+/**
+ * Function to parse a gVCF line into a SimpleVariant object
+ */
+private fun parseSingleGVCFLine(currentLine: String, samples: List<String>): SimpleVariant {
```
[the remark this answers is not in the document]
Comment (tcasstevens) [reply src/main/kotlin/biokotlin/util/VCFUtils.kt]: Ok, I added the VCF headers

File: src/main/kotlin/biokotlin/util/FileIO.kt [hunk 5]
```diff
@@ -23,4 +23,18 @@ fun bufferedReader(filename: String): BufferedReader {
...
+        println("bufferedWriter: ${e.message}")
```
[the remark this answers is not in the document]
Comment (tcasstevens) [reply src/main/kotlin/biokotlin/util/FileIO.kt:37]: Is there a logger for BioKotlin?

[the remark this answers is not in the document]
Comment (lynnjo) [reply src/main/kotlin/biokotlin/util/FileIO.kt:37]: Terry,Brandon - there is no official logger. but there are a few classes (MafPRocessingUtils.kt and Graph.kt) that use import io.github.oshai.kotlinlogging.KotlinLogging.  We should probably make a decision on this.

**Review by lynnjo — APPROVED** [review]

**Pull request merged**